pylearn2/models/mlp.py

```diff
@@ -3246,9 +3301,19 @@ def fprop(self, state_below):
                                                        "pooling.")
 
             if self.pool_type == 'max':
-                p = max_pool(bc01=d, pool_shape=self.pool_shape,
-                             pool_stride=self.pool_stride,
-                             image_shape=self.detector_space.shape)
+                # TODO Can they be merged?
+                if cudnn2d:
+                    # CPU implementation
+                    # TODO check if this can be ported to the GPU one
+                    # TODO check axes: should they be B01C or BC01 or will this
+                    # always work?
+                    p = downsample.max_pool_2d(d, self.pool_shape,
```

Comment:
Great catch! :) 
Actually, does it even make sense to use `downsample.max_pool_2d` instead of `max_pool` as in the "non cudnn2d" case? Apparently there is no reason. I am not sure if the data will be in bc01 format, though I think that the data has been already reshaped in that form somewhere since the "normal" convolution does not force you to provide data in bc01 format. If you don't already know the answer, I will check if this is the case.